src/utils/sourcemaps.rs

```diff
@@ -797,12 +797,14 @@ impl SourceMapProcessor {
 
         let api = Api::current();
 
-        if let Ok(artifacts) = api.list_release_files_by_checksum(
-            context.org,
-            context.project,
-            release,
-            &sources_checksums,
-        ) {
+        if let Ok(Ok(artifacts)) = api.authenticated().map(|api| {
```

Comment:
This can be written using `and_then`.
```suggestion
        if let Ok(artifacts) = api.authenticated().and_then(|api| {
```

One minor blemish here is that the function silently returns the wrong thing if you're not authenticated. In practice I believe it doesn't matter because the upload itself will surely fail in that case.